lazyflow/operators/ioOperators/opStreamingHdf5Reader.py

```diff
@@ -126,7 +133,14 @@ def execute(self, slot, subindex, roi, result):
         if result.flags.c_contiguous:
             hdf5File[internalPath].read_direct( result[...], key )
         else:
-            result[...] = hdf5File[internalPath][key]
+            if len(hdf5File[internalPath].shape) == 2:
+                roi.start = (roi.start[0], roi.start[1])
+                roi.stop = (roi.stop[0], roi.stop[1])
+                key = roi.toSlice()
+                result = result[:, :, 0]
```

Comment:
I wonder that this works, doesn't that change the shape of result? In any case you should rather do something like the following, and really return the result in the right shape (with an added channel axis)

```
tmp_result = hdf5File[internalPath][key]
result[...] = tmp_result[..., numpy.newaxis]
```

Reply:
yes it works but i am changing with this one. thanks